src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -543,8 +543,10 @@ private String key() {
                         }
                     } else {   // Execution nulled due to a critical failure, explicitly mark completed
                         completed = Boolean.TRUE;
+                        needsToPersist = true;  // Make sure we save toggled state
                     }
-                } else if (execution == null) {
+                } else if (execution == null && completed != Boolean.TRUE) {
+                    needsToPersist = true; // Make sure we save toggled state
```

Comment:
Not really related to changes in JEP-210, right?

Reply:
Not as such, but more correcting for some issues in the logic.